docs/InstallGuide.md

```diff
@@ -1,424 +1,358 @@
+This means the OpenFlow controller which is vna is now connected to the datapath. After the connection between the OpenFlow controller and the
+datapath is established it starts installing the flows on the datapath.
 
+Verification
+-------
 
-Check Virtual Network Operation.
-------------------------------------
+By following all the instructions from the beginning to this section, you already have the sample environment. In this section, we are firstly going to play around the environment then secondary see network packets going through OpenFlow rules to reach
+from one guest to another.
 
-### Start VM(s)
+As a representation of the guest here we use [LXC](https://linuxcontainers.org), which helps users run multiple isolated Linux system containers. Any other virtualization technologies might be used, however we take LXC since it is easy to create/destroy/configure and simple enough to do this verification.
 
-We use kvm to start four virtual machines. Their interfaces were registered in the database earlier. These VMs and their networks are arranged as follows.
+Install LXC
 
-+ vm1, vm3 -> vnet1
-+ vm2, vm4 -> vnet2
+```
+# yum -y install lxc lxc-templates
+```
 
-The command "run-vm-all" starts all four virtual machines at once.
+Create and mount cgroup
 
-    # cd /root/vnet-test-kvm/vm
-    # ./run-vm-all
-    *** vm1 ***
-    node-id: 1
-    interface name: if-x5vjmksl
-    mac address: 52:54:00:0d:84:8c
-    *** vm2 ***
-    node-id: 2
-    interface name: if-sve1s6xp
-    mac address: 52:54:00:9a:cc:00
-    *** vm3 ***
-    node-id: 3
-    interface name: if-bvixz40n
-    mac address: 52:54:00:0a:fb:78
-    *** vm4 ***
-    node-id: 4
-    interface name: if-7st2alal
-    mac address: 52:54:00:e5:f9:fb
+```
+# mkdir /cgroup
+# echo "cgroup /cgroup cgroup defaults 0 0" >> /etc/fstab
+# mount /cgroup
+```
 
-### Check the IP address of virtual machines
+Create 2 LXC guests
+(it may require rsync to be installed)
 
-You can log into the virtual machines using the telnet command.
+```
+# lxc-create -t centos -n inst1
+# lxc-create -t centos -n inst2
+```
 
-    # telnet localhost PORT-NO
+Configure interfaces of each guest
 
-The port numbers are:
+```
+# vi /var/lib/lxc/inst1/config
+lxc.network.type = veth
+lxc.network.flags = up
+lxc.network.veth.pair = inst1
+lxc.network.ipv4 = 10.100.0.10
+lxc.network.hwaddr = 10:54:FF:00:00:01
+lxc.rootfs = /var/lib/lxc/inst1/rootfs
+lxc.include = /usr/share/lxc/config/centos.common.conf
+lxc.arch = x86_64
+lxc.utsname = inst1
+lxc.autodev = 0
+```
 
-+ vm1 : 5001
-+ vm2 : 5002
-+ vm3 : 5003
-+ vm4 : 5004
+```
+# vi /var/lib/lxc/inst2/config
+lxc.network.type = veth
+lxc.network.flags = up
+lxc.network.veth.pair = inst2
+lxc.network.ipv4 = 10.100.0.11
+lxc.network.hwaddr = 10:54:FF:00:00:02
+lxc.rootfs = /var/lib/lxc/inst2/rootfs
+lxc.include = /usr/share/lxc/config/centos.common.conf
+lxc.arch = x86_64
+lxc.utsname = inst2
+lxc.autodev = 0
+```
```

Comment:
I think it's also a good idea to make a remark here that the `lxc.network.link = br0` option can not be used to enslave the containers' network interfaces to Open vSwitch. People familiar with the Linux bridge might try this option but it will crash when looking for the Linux bridge and finding Open vSwitch instead.